prometheus/go_collector.go

```diff
@@ -236,6 +242,9 @@ func (c *goCollector) Describe(ch chan<- *Desc) {
 func (c *goCollector) Collect(ch chan<- Metric) {
 	c.goroutines.Set(float64(runtime.NumGoroutine()))
 	ch <- c.goroutines
+	n, _ := runtime.ThreadCreateProfile(nil)
```

Comment:
I did some consistency checks. The threadcreate number indeed looks like the number of OS threads.